scripts/Game.gd

```diff
@@ -12,4 +12,36 @@ var levels = {
         "name": "Second level",
     }
 }
-var current_level = null
+var current_level_id = null
+
+func get_current_level():
+    return levels[current_level_id]
+
+var save_slots = [
```

Comment:
Does this save the max amount of coins the player has collected from that level during all playthroughs? This is more of a game design question I think.

There are other possibilities, like tracking which coins were picked up... but that leads to persisting the state of every visited level... not only coins, but the state of enemies, etc...

The choice made here is a reasonable one, if we want to make an episodic game with re-playable levels (do we?). It is certainly simpler to implement 😄

Reply:
Explanation to this is in comment in main thread.